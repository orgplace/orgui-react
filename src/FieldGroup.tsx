import classNames from 'classnames';
import React from 'react';

interface OuiFieldGroupAttributes {
  label: React.ReactChild,
  required?: boolean,
  hazarding?: boolean,
  helpText?: React.ReactChild,
}

class OuiFieldGroup extends React.PureComponent<React.HTMLAttributes<HTMLElement> & OuiFieldGroupAttributes> {
  render() {
    const {label, required, hazarding, helpText, ...divProps} = this.props;
    return (
      <div className={classNames(
        'ouiFieldGroup',
        {
          'isRequired': required,
          'isHazarding': hazarding,
        },
        divProps.className,
      )}>
        <label className='ouiFieldGroup__label'>{label}</label>
        {this.renderChildren()}
        {this.renderHelpText()}
      </div>
    );
  }

  renderChildren() {
    const child = React.Children.only(this.props.children) as React.ReactElement;
    return React.cloneElement(child, {
      className: classNames(child.props.className, 'ouiFieldGroup__child'),
    });
  }

  renderHelpText() {
    const helpText = this.props.helpText;
    if (!helpText) {
      return null;
    }

    if (typeof helpText === 'string' || typeof helpText === 'number') {
      return (
        <div className='ouiFieldGroup__helpText'>help</div>
      );
    } else {
      return React.cloneElement(helpText, {
        className: classNames(helpText.props.className, 'ouiFieldGroup__helpText'),
      });
    }
  }
}


export { OuiFieldGroup };
