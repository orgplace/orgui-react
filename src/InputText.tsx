import classNames from 'classnames';
import React from 'react';

const OuiInputText: React.FC<React.InputHTMLAttributes<HTMLInputElement>> = (props) => {
  return (
    <input {...props} type="text" className={classNames(
      'ouiInputText',
      props.className,
    )} />
  );
}

export { OuiInputText };
