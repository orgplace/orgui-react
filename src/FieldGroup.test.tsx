import React from 'react';
import renderer from 'react-test-renderer';
import { OuiFieldGroup } from './FieldGroup';
import { OuiInputText } from './InputText';

test('OuiFieldGroup', function() {
	const component = renderer.create(
		<OuiFieldGroup label={<span style={{fontWeight: "bold"}}>Label</span>} required helpText="help">
			<OuiInputText />
		</OuiFieldGroup>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
