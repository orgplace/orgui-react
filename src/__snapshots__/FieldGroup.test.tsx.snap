// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiFieldGroup 1`] = `
<div
  className="ouiFieldGroup isRequired"
>
  <label
    className="ouiFieldGroup__label"
  >
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Label
    </span>
  </label>
  <input
    className="ouiInputText ouiFieldGroup__child"
    type="text"
  />
  <div
    className="ouiFieldGroup__helpText"
  >
    help
  </div>
</div>
`;
